src/Workspaces/Core/Portable/CodeFixes/FixAllOccurrences/TextChangeMerger.cs

```diff
@@ -0,0 +1,232 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Shared.Collections;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.CodeFixes
+{
+    /// <summary>
+    /// Helper to merge many disparate text changes to a single document together into a total set of changes.
+    /// </summary>
+    internal class TextChangeMerger
+    {
+        private readonly struct IntervalIntrospector : IIntervalIntrospector<TextChange>
+        {
+            int IIntervalIntrospector<TextChange>.GetStart(TextChange value) => value.Span.Start;
+            int IIntervalIntrospector<TextChange>.GetLength(TextChange value) => value.Span.Length;
+        }
+
+        private readonly Document _oldDocument;
+        private readonly IDocumentTextDifferencingService _differenceService;
+
+        private readonly SimpleIntervalTree<TextChange, IntervalIntrospector> _totalChangesIntervalTree =
+            SimpleIntervalTree.Create(new IntervalIntrospector(), Array.Empty<TextChange>());
+
+        public TextChangeMerger(Document document)
+        {
+            _oldDocument = document;
+            _differenceService = document.Project.Solution.Workspace.Services.GetRequiredService<IDocumentTextDifferencingService>();
+        }
+
+        /// <summary>
+        /// Try to merge the changes made to <paramref name="newDocument"/> into the tracked changes. If there is any
+        /// conflicting change in <paramref name="newDocument"/> with existing changes, then no changes are added.
+        /// </summary>
+        public async Task TryMergeChangesAsync(Document newDocument, CancellationToken cancellationToken)
```

Comment:
all this logic moved over here unchanged *except* that the new code now accesses instance fields for some constant values instead of needing them passed in to each call.